### PR TITLE
Update `CardTheme`, `DrawerTheme`, `NavigationBar`, and `NavigationRailTheme` tests for M2/M3 

### PR DESCRIPTION
**Pull request by QuncCccccc**
Updated unit tests for `CardTheme`, `DrawerTheme`, `NavigationBar`, and `NavigationRailTheme` to have M2 and M3 versions.

More info in https://github.com/flutter/flutter/issues/127064

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by HansMuller** [comment]
Currently there are some test failures, e.g.
```
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
 The following TestFailure was thrown running a test:
 Expected: MaterialColor:<MaterialColor(primary value: Color(0xff2196f3))>
   Actual: Color:<Color(0xff6750a4)>
 When the exception was thrown, this was the stack:
 #4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart:154:5)
 <asynchronous suspension>
 <asynchronous suspension>
 (elided one frame from package:stack_trace)
 This was caught by the test expectation on the following line:
   file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart line 154
 The test description was:
   Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in
   end drawer
 ════════════════════════════════════════════════════════════════════════════════════════════════════
 06:46 +7906 ~23 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart: Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in end drawer [E]
   Test failed. See exception logs above.
   The test description was: Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in end drawer
   
```

**Comment by QuncCccccc** [comment]
> Currently there are some test failures, e.g.
> 
> ```
>  ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
>  The following TestFailure was thrown running a test:
>  Expected: MaterialColor:<MaterialColor(primary value: Color(0xff2196f3))>
>    Actual: Color:<Color(0xff6750a4)>
>  When the exception was thrown, this was the stack:
>  #4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart:154:5)
>  <asynchronous suspension>
>  <asynchronous suspension>
>  (elided one frame from package:stack_trace)
>  This was caught by the test expectation on the following line:
>    file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart line 154
>  The test description was:
>    Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in
>    end drawer
>  ════════════════════════════════════════════════════════════════════════════════════════════════════
>  06:46 +7906 ~23 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/drawer_theme_test.dart: Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in end drawer [E]
>    Test failed. See exception logs above.
>    The test description was: Material3 - Default values are used when no Drawer or DrawerThemeData properties are specified in end drawer
>    
> ```

Thanks for letting me know! I fixed the error yesterday but seems one test can pass locally but not on Github again([link](00:40 +101: test/material/navigation_bar_test.dart: Material3 - NavigationBar shows tooltips with text scaling                                                                                         
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: Size:<Size(15.0, 20.0)>
  Actual: _DebugSize:<Size(15.0, 21.0)>
)). Taking a look.

**Review by TahaTesser — APPROVED** [review]
LGTM!

File: packages/flutter/test/material/navigation_bar_test.dart [hunk 1]
```diff
@@ -305,7 +305,7 @@ void main() {
     expect(_getIndicatorDecoration(tester)?.shape, const StadiumBorder());
   });
 
-  testWidgets('NavigationBar shows tooltips with text scaling ', (WidgetTester tester) async {
+  testWidgets('Material2 - NavigationBar shows tooltips with text scaling ', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/navigation_bar_test.dart:308]:
```suggestion
  testWidgets('Material2 - NavigationBar shows tooltips with text scaling', (WidgetTester tester) async {
```

File: packages/flutter/test/material/navigation_bar_test.dart [hunk 2]
```diff
@@ -364,6 +364,72 @@ void main() {
     expect(tester.getSize(find.text(label).last), Size(defaultTooltipSize.width * 4, defaultTooltipSize.height * 4));
   });
 
+  testWidgets('Material3 - NavigationBar shows tooltips with text scaling ', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/navigation_bar_test.dart:367]:
```suggestion
  testWidgets('Material3 - NavigationBar shows tooltips with text scaling', (WidgetTester tester) async {
```

**Pull request merged**